--- imsettings-1.4.0/imsettings/imsettings-utils.c~	2012-07-12 10:22:21.000000000 +0200
+++ imsettings-1.4.0/imsettings/imsettings-utils.c	2012-11-12 21:51:48.147954848 +0100
@@ -146,7 +146,7 @@
 		GDBusNodeInfo *introspection_data = g_dbus_node_info_new_for_xml(introspection_xml, &err);
 
 		if (err) {
-			g_warning(err->message);
+			g_warning("%s", err->message);
 			return NULL;
 		}
 		info = g_dbus_interface_info_ref(introspection_data->interfaces[0]);
--- imsettings-1.4.0/imsettings-daemon/imsettings-module.c~	2011-10-14 05:23:26.000000000 +0200
+++ imsettings-1.4.0/imsettings-daemon/imsettings-module.c	2012-11-12 21:52:35.441286514 +0100
@@ -260,7 +260,7 @@
 						&mod_cb);
 
 				if (!mod_cb) {
-					g_warning(g_module_error());
+					g_warning("%s", g_module_error());
 					goto next;
 				}
 				priv->switch_im = mod_cb;
@@ -268,7 +268,7 @@
 						"module_dump_im",
 						&mod_cb);
 				if (!mod_cb) {
-					g_warning(g_module_error());
+					g_warning("%s", g_module_error());
 					goto next;
 				}
 				priv->dump_im = mod_cb;
--- imsettings-1.4.0/imsettings-daemon/imsettings-proc.c~	2012-04-26 04:15:38.000000000 +0200
+++ imsettings-1.4.0/imsettings-daemon/imsettings-proc.c	2012-11-12 21:53:11.314618581 +0100
@@ -109,7 +109,7 @@
 	}
 	str = g_string_new(NULL);
 	g_string_append_printf(str, "%s[%lu]: %s", module, (gulong)priv->main.pid, buffer);
-	g_message(str->str);
+	g_message("%s", str->str);
 	g_string_free(str, TRUE);
 	g_free(buffer);
 
@@ -204,7 +204,7 @@
 							 type_names[type],
 							 module);
 
-			g_critical(message);
+			g_critical("%s", message);
 			g_signal_emit(proc, signals[SIG_NOTIFY_NOTIFICATION], 0, NOTIFY_URGENCY_CRITICAL, title, message, 0);
 			g_free(message);
 			unref = TRUE;
@@ -303,7 +303,7 @@
 		g_set_error(error, IMSETTINGS_GERROR, IMSETTINGS_GERROR_UNKNOWN,
 			    _("[BUG] %s process is still running [pid: %d]\n"),
 			    type_name, pinfo->pid);
-		g_warning((*error)->message);
+		g_warning("%s", (*error)->message);
 		return FALSE;
 	} else {
 		/* close io channels prior to create new one */
--- imsettings-1.4.0/backends/lxde/lxde-module.c~	2012-04-23 10:28:00.000000000 +0200
+++ imsettings-1.4.0/backends/lxde/lxde-module.c	2012-11-12 21:54:41.767948722 +0100
@@ -176,7 +176,7 @@
 
 	retval = g_key_file_get_string(key, "GTK", "sGtk/IMModule", &err);
 	if (err) {
-		g_warning(err->message);
+		g_warning("%s", err->message);
 		g_error_free(err);
 	}
   finalize:
--- imsettings-1.4.0/backends/xim/loopback.c~	2011-03-08 07:39:32.000000000 +0100
+++ imsettings-1.4.0/backends/xim/loopback.c	2012-11-12 21:56:22.811278492 +0100
@@ -774,7 +774,7 @@
 		gboolean retval;
 
 		g_xim_messages_warning(G_XIM_PROTOCOL_GET_IFACE (proto)->message,
-				       msg);
+				       "%s", msg);
 		retval = g_xim_connection_cmd_error(G_XIM_CONNECTION (proto),
 						    imid, icid, G_XIM_EMASK_VALID_IMID | G_XIM_EMASK_VALID_ICID,
 						    G_XIM_ERR_BadProtocol,
@@ -804,7 +804,7 @@
 		gboolean retval;
 
 		g_xim_messages_warning(G_XIM_PROTOCOL_GET_IFACE (proto)->message,
-				       msg);
+				       "%s", msg);
 		retval = g_xim_connection_cmd_error(G_XIM_CONNECTION (proto),
 						    imid, icid, G_XIM_EMASK_VALID_IMID | G_XIM_EMASK_VALID_ICID,
 						    G_XIM_ERR_BadProtocol,
@@ -845,7 +845,7 @@
 		gboolean retval;
 
 		g_xim_messages_warning(G_XIM_PROTOCOL_GET_IFACE (proto)->message,
-				       msg);
+				       "%s", msg);
 		retval = g_xim_connection_cmd_error(G_XIM_CONNECTION (proto),
 						    imid, icid, G_XIM_EMASK_VALID_IMID | G_XIM_EMASK_VALID_ICID,
 						    G_XIM_ERR_BadProtocol,
--- imsettings-1.4.0/backends/xim/proxy.c~	2011-03-08 07:39:32.000000000 +0100
+++ imsettings-1.4.0/backends/xim/proxy.c	2012-11-12 21:56:58.437943903 +0100
@@ -796,7 +796,7 @@
 		retval = g_xim_connection_cmd_error(G_XIM_CONNECTION (conn), imid, icid, flag,
 						    G_XIM_ERR_BadProtocol, 0, msg);
 	}
-	g_xim_messages_error(G_XIM_PROTOCOL_GET_IFACE (proto)->message, msg);
+	g_xim_messages_error(G_XIM_PROTOCOL_GET_IFACE (proto)->message, "%s", msg);
 	g_free(msg);
 
 	return retval;
